builtin/providers/aws/resource_aws_elasticache_cluster.go

```diff
@@ -361,9 +364,15 @@ func resourceAwsElasticacheClusterRead(d *schema.ResourceData, meta interface{})
 		}
 
 		d.Set("subnet_group_name", c.CacheSubnetGroupName)
-		d.Set("security_group_names", c.CacheSecurityGroups)
-		d.Set("security_group_ids", c.SecurityGroups)
-		d.Set("parameter_group_name", c.CacheParameterGroup)
+		if len(c.CacheSecurityGroups) > 0 {
+			d.Set("security_group_names", flattenElastiCacheSecurityGroupNames(c.CacheSecurityGroups))
+		}
+		if len(c.SecurityGroups) > 0 {
+			d.Set("security_group_ids", flattenElastiCacheSecurityGroupIds(c.SecurityGroups))
+		}
+		if c.CacheParameterGroup != nil {
+			d.Set("parameter_group_name", c.CacheParameterGroup.CacheParameterGroupName)
+		}
```

Comment:
We should be safe to simply do each of these `set` operations with the results of the respective `flatten` methods. By only doing so if `len > 0`, we may fail to detect drift. Ex. we should have a single SG, but an operator inadvertently  removed it in the Web UI; TF would fail to detect that